kong/db/schema/entities/certificates.lua

```diff
@@ -11,13 +12,18 @@ return {
     { id = typedefs.uuid, },
     { created_at     = typedefs.auto_timestamp_s },
     { cert           = typedefs.certificate { required = true }, },
-    { key            = typedefs.key         { required = true }, },
+    { key            = typedefs.key, },
   },
 
   entity_checks = {
     { custom_entity_check = {
       field_sources = { "cert", "key" },
       fn = function(entity)
+        if not entity.key or entity.key == null then
```

Comment:
`if entity.key == null` should be sufficient; this function is not called on a partial update if `cert` and `key` are not given.